server.go

```diff
@@ -824,17 +824,25 @@ func (s *Server) Name() protofsm.EndpointName {
 // CanHandle returns true if the target message can be routed to this endpoint.
 //
 // NOTE: This method is part of the protofsm.MsgEndpoint interface.
-func (s *Server) CanHandle(msg protofsm.PeerMsg) bool {
-	<-s.ready
+func (s *Server) CanHandle(msg protofsm.PeerMsg) (bool, error) {
+	err := s.waitForReady()
```

Comment:
Depending on the discussion on the `lnd` PR it _might_ be okay to just return `false` if `s.waitForReady()` returns an error. But calling `s.waitForReady()` should definitely be done so we don't block indefinitely.

Reply:
Right, we don't really want unnecessary error decorations in this interface.

Instead can just log a debug msg and return false